docpages/make_a_bot/cmake.md

```diff
@@ -1,74 +1,87 @@
-\page buildcmake Building a Discord Bot using CMake/UNIX
+\page buildcmake Building a Discord Bot Using CMake (UNIX)
 
 ## 1. Toolchain
-Before compiling, you will need to install `cmake` on your system.
-To be sure that `cmake` is installed, you can type the following command:
-
-    $ cmake --version
-    cmake version 3.20.4
+Before compiling, you will need to install `cmake` on your system. To be sure that `cmake` is installed, you can type the following command:
 
+```bash
+$ cmake --version
+cmake version 3.22.1
+```
 
 ## 2. Create a CMake project
 
 In an empty directory, create the following files and directories:
 
-    - your_project/
-        |-- libs/
-        |-- src/
-            |-- main.cpp
-        |-- CMakeLists.txt
-
-
-In the `libs/` directory, clone D++ with: `git clone https://github.com/brainboxdotcc/DPP.git`
+```puml
+- your_project/
+    |-- build/
+	|-- libs/
+	|-- src/
+		|-- main.cpp
+	|-- CMakeLists.txt
+```
+
+Now, you'll want to clone D++ in the `libs/` directory. You can clone D++ with: `git clone https://github.com/brainboxdotcc/DPP.git`.
+
+Once that's done, your project directory should look like this:
+
+```puml
+- your_project/
+    |-- build/
+	|-- libs/
+		|-- DPP/
+	|-- src/
+		|-- main.cpp
+	|-- CMakeLists.txt
+```
 
 ## 3. Configure CMake
 
-Here is an example CMake configuration, adapt it according to your needs:
+You'll need to modify the `CMakeLists.txt` to tell CMake what it's looking for, and other information.
 
-~~~~~~~~~~~~~~{.cmake}
+Here is an example CMake configuration, you can adapt it according to your needs:
+
+~~~~~~~~~~~~~~cmake
 # minimum CMake version required
 cmake_minimum_required(VERSION 3.15)
 # Project name, version and description
-project(discord-bot VERSION 1.0 DESCRIPTION "A discord bot")
+project(discord-bot VERSION 1.0 DESCRIPTION "A Discord bot")
 
 # Add DPP as dependency
 add_subdirectory(libs/DPP)
```

Comment:
this whole approach of having the lib in a subdirectory conflicts with the "install it systemwide" tutorials we have before this. Can we change it to assume its built systemwide? it makes everything so much easier and packaging libs in a project is awful.

Reply:
Right, so the only docs that should cover building from source are those under "Building D++ From Source"? The rest of the docs should be assuming DPP is installed?

Reply:
yes that sounds right

Reply:
remember you can draw these in dia, then paste the text of the saved .dot file in